src/combination.jl

```diff
@@ -1,37 +1,43 @@
-""" This file contains all the functions related to Bliss and temporal combinations. """
+""" This file contains all the functions related to Bliss combinations. """
 
-function BlissCombination(p1::Array{Float64, 2}, p2::Array{Float64, 2}, n::Int)
+######---------------- Functions for Bliss combination ----------------########
+function CombinationParam(p1::Array{Float64, 2}, p2::Array{Float64, 2}, n::Int)
     """ A function to calculate Bliss independence for drug combination assuming
     the two drugs hit different pathways and they effect independently. """
 
+    pprime1 = copy(p1)
+    pprime2 = copy(p2)
     param1 = zeros(4, 8)
     param2 = zeros(4, 8)
-    param1[1, :] .= 1.0 .- (p1[1, :] ./ p1[1, 1])
-    param1[2, :] .= 1.0 .- (p1[2, :] ./ p1[2, 1])
-    param1[3, :] .= p1[3, :]
-    param1[4, :] .= p1[4, :]
-    param2[1, :] .= 1.0 .- (p2[1, :] ./ p2[1, 1])
-    param2[2, :] .= 1.0 .- (p2[2, :] ./ p2[2, 1])
-    param2[3, :] .= p2[3, :]
-    param2[4, :] .= p2[4, :]
+    # drug A
+    param1[1:2, :] .= 1.0 .- (pprime1[1:2, :] ./ pprime1[1:2, 1]) # g1 and g2 prog. rates
+    param1[3:4, :] .= pprime1[3:4, :]                             # g1 and g2 death rates
+    # drug B
+    param2[1:2, :] .= 1.0 .- (pprime2[1:2, :] ./ pprime2[1:2, 1])
+    param2[3:4, :] .= pprime2[3:4, :]
+
+    # make sure normalized correctly; the prog. rates in control condition must be zero in both drugs.
+    @assert param1[1,1] == param1[2,1] == param2[1,1] == param2[2,1] == 0.0
 
     """ For 8x8 combination of drug concentrations for G1 progression rate, G2 progression rate, and death rates in G1 and G2, respectively. """
     combined = zeros(n, n, 4)
     for j = 1:n
-        for k = 1:n
-            combined[j, k, 1:2] .= (1.0 .- (param1[1:2, j] .+ param2[1:2, k] .- param1[1:2, j] .* param2[1:2, k])) .* p1[1:2, 1, 1]
+        for k = 1:n # param2 is changing, param1 is constant
+            combined[j, k, 1:2] .= (1.0 .- (param1[1:2, j] .+ param2[1:2, k] .- param1[1:2, j] .* param2[1:2, k])) .* ((p1[1:2, 1] .+ p2[1:2, 1]) ./ 2)
             combined[j, k, 3:4] .= param1[3:4, j] .+ param2[3:4, k]
         end
     end
```

Comment:
I thought my indexing might be wrong like I had flipped some `j` or `k` but I wrote it down for a few columns and rows and it seemed right.